src/OrchardCore/OrchardCore.Roles.Core/DefaultSystemRoleProvider.cs

```diff
@@ -0,0 +1,70 @@
+using System.Collections.Frozen;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Options;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Security;
+
+namespace OrchardCore.Roles;
+
+public sealed class DefaultSystemRoleProvider : ISystemRoleProvider
+{
+    private readonly FrozenDictionary<string, Role> _systemRoles;
+    private readonly Role _adminRole;
+    private readonly IStringLocalizer S;
+
+    public DefaultSystemRoleProvider(
+        ShellSettings shellSettings,
+        IStringLocalizer<DefaultSystemRoleProvider> stringLocalizer,
```

Comment:
```suggestion
        IStringLocalizer<DefaultSystemRoleProvider> S,
```

Reply:
This previous one already suggested by you :)

Reply:
Leave it as it is, because PO Extractor might not support this

Reply:
It's does. And we use it all the time. There is no need to maintain a local instance since we don't use

Reply:
I will TRUST you one this, coz I'm not sure if it's supported